src/System.CommandLine.Hosting/HostingExtensions.cs

```diff
@@ -81,5 +81,40 @@ public static class HostingExtensions
                 modelBinder.UpdateInstance(opts, bindingContext);
             });
         }
+
+        public static IHostBuilder UseCommandHandler<TCommand, THandler>(this IHostBuilder builder)
+            where TCommand : Command
+            where THandler : ICommandHandler
+        {
+            return builder.UseCommandHandler(typeof(TCommand), typeof(THandler));
+        }
+
+        public static IHostBuilder UseCommandHandler(this IHostBuilder builder, Type commandType, Type handlerType)
+        {
+            if (!typeof(Command).IsAssignableFrom(commandType))
+            {
+                throw new ArgumentException($"{nameof(commandType)} must be a type of {nameof(Command)}", nameof(handlerType));
+            }
+
+            if (!typeof(ICommandHandler).IsAssignableFrom(handlerType))
+            {
+                throw new ArgumentException($"{nameof(handlerType)} must implement {nameof(ICommandHandler)}", nameof(handlerType));
+            }
+
+            if (builder.Properties[typeof(InvocationContext)] is InvocationContext invocation 
+                && invocation.ParseResult.CommandResult.Command is Command command
+                && command.GetType() == commandType)
```

Comment:
I am unsure if it risks causing more confusion if a command and a more specific command is registered, then the handler for the command might end up matching with the more specific command if i am understanding the scenario correctly.

Reply:
It looks like it might also be redundant with the following line. 

Is there a way to ask the container if it can create an instance of `handlerType`, so we don't implement logic that's inconsistent with the container configuration?

Reply:
Are you thinking testing if the container already has the handler type registered? Or verifying we are not parsed something the container cannot possibly new up?

Reply:
For example, the container might be configured to map `ISomeCommand` to `MyCommand`. The current code would block that from working because of the `command.GetType() == commandType` check. So in effect we're reimplementing a type matching check event though the resolved handler might work just fine.

Reply:
The problem is that the check is there to ensure that the command is the one that the caller configured to be handeled by that commandHandler.

Is it even possible to have the command as something not inheriting from `Command` like an interface?